packages/docs-reanimated/docs/debugging/accurate-call-stacks.mdx

```diff
@@ -0,0 +1,39 @@
+---
+id: accurate-call-stacks
+sidebar_label: Accurate Call Stacks
+sidebar_position: 1
+---
+
+# Accurate Call Stacks
+
+When debugging Reanimated code, you may encounter error or warning call stacks that don’t clearly indicate the root cause of a problem. This occurs because, by default, React Native displays the entire call stack up to the point where the error was thrown or the warning was logged. This can be misleading, as it often highlights code from Reanimated internals instead of the specific problematic code within your project.
+
+To address this, Metro provides a way to enhance stack traces through the Metro `symbolicator`. Reanimated utilizes this feature by exporting a Metro config wrapper called `wrapWithReanimatedMetroConfig`. This wrapper configures the symbolicator to improve the accuracy of stack traces, making it easier to identify issues within your code.
```

Comment:
While this is well written, I'm afraid it might not address its target audience properly.

First of all, these paragraphs could suggest like we're moving the responsibility of knowing whether an error comes from Reanimated or the user's code to the user himself with little guidance.

Secondly, there's no point at which Reanimated users would be incentivized to look-up this page in the docs. In my opinion they'd still be stuck on vague errors, googling them or creating new issues.

Ideally, we'd have the wrapper enabled by default, but I guess it isn't really feasible, analogous to the case of Reanimated Babel plugin.

To solve this issue, I have a few ideas:
- Let's put the information about the Metro Wrapper somewhere in the installation guide as an __recommended, optional step__.
- If it's possible in runtime to obtain the information whether the Metro Wrapper is present, we could embed some runtime solution. For example, we could add a one-time warning during Reanimated initialization that Metro Wrapper isn't present or we could append information about the Wrapper to every error thrown by Reanimated.

I think we should consider who exactly the target of our Metro Wrapper is, probably specify it in these docs and edit them accordingly.

As a side note, I think technical details here might be unnecessarily intertwined with DX-relevant information. Let's make sure the most important information don't include technicalities and provide them later on.

Reply:
> If it's possible in runtime to obtain the information whether the Metro Wrapper is present, we could embed some runtime solution. For example, we could add a one-time warning during Reanimated initialization that Metro Wrapper isn't present or we could append information about the Wrapper to every error thrown by Reanimated.

I'm afraid this won't be possible. We don't have a direct access to metro from react native so we likely won't be able to detect if this config is used.

Even though we pass an error stack trace in out custom logger to the `addLog` method from `LogBox`, this stack trace is not symbolicated and we don't know how it will look after metro symbolication is applied (and whether our custom metro symbolicator is used).

> Let's put the information about the Metro Wrapper somewhere in the installation guide as an recommended, optional step.

Will this look fine or should I change something?

![Screenshot 2024-09-02 at 13 37 36](https://github.com/user-attachments/assets/2e369876-27fb-4aef-aef3-f058ce29e213)

Reply:
> I think we should consider who exactly the target of our Metro Wrapper is, probably specify it in these docs and edit them accordingly.

I think that this config wrapper will be helpful for everyone, also for devs. Do you think that we wouldn't want to recommend it for everyone? I think that most developers, who use Reanimated as a dependency of their library, would prefer to see detailed errors with proper code snippet.

If you meant developers of Reanimated, we can still unwrap wrapped stack frames. They aren't removed, so adding this wrapper won't remove anything form stack. These errors would be helpful even for us, e.g. while developing an example app.